src/method_ilan.jl

```diff
@@ -166,32 +172,71 @@ function ilan(
 
         orthogonalize_and_normalize!(view(V,:,1:k),view(V,:,k+1), view(HH,1:k,k), orthmethod)
 
+
         # extract eigenpair approximation
         if ((rem(k,check_error_every)==0)||(k==m))&&(k>2)
-            VV=view(V,:,1:k+1)
+            if !proj_solve
+
+                # Extract eigenvalues from Hessenberg matrix
+                D,W_Ritz = eigen(H[1:k,1:k])
+
+                mul!(view(W,:,1:k),view(QQ,:,1:k),W_Ritz)
+                #ZZ = view(QQ,:,1:k)*Z_Ritz
+                λ = σ .+ γ ./ D
+            else
+                VV=view(V,:,1:k+1)
+                # Projected solve to extract eigenvalues (otw hessenberg matrix)
+
+                # create the projected NEP
+                mm=size(VV,2)
+                pnep=create_proj_NEP(nep,mm); # maxsize=mm
+                set_projectmatrices!(pnep,VV,VV);
+
+                # solve the projected NEP
+                push_info!(logger,2,"Solving the projected problem",continues=true);
 
-            # create the projected NEP
-            mm=size(VV,2)
-            pnep=create_proj_NEP(nep,mm); # maxsize=mm
-            set_projectmatrices!(pnep,VV,VV);
-            err_lifted=(λ,z)->estimate_error(errmeasure,λ,VV*z);
+                λproj,Wproj=inner_solve(inner_solver_method,T,pnep;
+                                        neigs=Inf,
+                                        inner_logger=inner_logger);
 
-            # solve the projected NEP
-            push_info!(logger,2,"Solving the projected problem",continues=true);
-            λ,ZZ=iar(pnep;neigs=Inf,logger=0,maxit=150,tol=tol,check_error_every=Inf,errmeasure=err_lifted)
-            push_info!(logger,2,".");
-            W=VV*ZZ;
+                q=length(λproj)
+                if q>m q=m end
+                mul!(view(W,:,1:q),VV,view(Wproj,:,1:q))
+                #ZZ=VV*Zproj[:,1:k];
+                λ=λproj[1:q];
```

Comment:
Doesn't this line overwrite the allocated λ?  Maybe the  `λ=zeros(T,5m+1);` above can be removed